framework/src/utils/PetscSupport.C

```diff
@@ -632,6 +632,26 @@ getCommonPetscKeys()
     "-snes_ls -snes_max_it -snes_rtol -snes_type -sub_ksp_type -sub_pc_type", "", true);
 }
 
+void
+setSinglePetscOption(const char * name, const char * value)
```

Comment:
But then, we would need something special for things like `setSinglePetscOption("-snes_fd", PETSC_NULL);`...